src/services/cgnOperatorSearchService.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ * This service interacts with the GCN operator search API
+ */
+import {
+  IResponseErrorInternal,
+  IResponseErrorNotFound,
+  IResponseErrorValidation,
+  IResponseSuccessJson,
+  ProblemJson,
+  ResponseErrorInternal,
+  ResponseErrorNotFound,
+  ResponseSuccessJson
+} from "italia-ts-commons/lib/responses";
+
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
+import { Merchant } from "../../generated/cgn-operator-search/Merchant";
+import {
+  ResponseErrorStatusNotDefinedInSpec,
+  withCatchAsInternalError,
+  withValidatedOrInternalError
+} from "../utils/responses";
+import { readableProblem } from "../utils/errorsFormatter";
+import { CgnOperatorSearchAPIClient } from "../../src/clients/cgn-operator-search";
+import { OnlineMerchantSearchRequest } from "../../generated/io-cgn-operator-search-api/OnlineMerchantSearchRequest";
+import { OnlineMerchants } from "../../generated/cgn-operator-search/OnlineMerchants";
+import { OfflineMerchantSearchRequest } from "../../generated/io-cgn-operator-search-api/OfflineMerchantSearchRequest";
+import { OfflineMerchants } from "../../generated/cgn-operator-search/OfflineMerchants";
+import { IResponseType } from "italia-ts-commons/lib/requests";
+export default class CgnService {
+  constructor(
+    private readonly cgnOperatorSearchApiClient: ReturnType<
+      CgnOperatorSearchAPIClient
+    >
+  ) {}
+
+  private readonly toResponse = <T>(
```

Comment:
❤️

Reply:
This map is correct until the `T` type of `IResponse<200, T>` is equal to `IResponseSuccessJson<T>

Reply:
But that's ok for now, if specifications changes we will think about it.

Reply:
I think that it will be the most common case. If we need a change we'll make that change.

Reply:
The type is coerent with the map implementation. If the type of the response swill diverge from the IResponse, both the function type declaration and the map body itself should be modified. For example:
```ts
private readonly toResponse = <T,R>(
response:
      | IResponseType<200, T>
      | IResponseType<404, undefined>
      | IResponseType<500, ProblemJson>,
map: (a: T) => R 
  ):
    | IResponseErrorInternal
    | IResponseErrorValidation
    | IResponseErrorNotFound
    | IResponseSuccessJson<R> =>
....
      return ResponseSuccessJson(map(response.value));
```